components/ai_chat/core/browser/conversation_handler.h

```diff
@@ -60,12 +60,22 @@ class AIChatService;
 class AssociatedArchiveContent;
 class AIChatCredentialManager;
 
+class ConversationHandlerForMetrics {
+ public:
+  virtual ~ConversationHandlerForMetrics() = default;
+  virtual bool IsConversationStarter(std::string_view title) = 0;
+  virtual size_t GetConversationHistorySize() = 0;
+  virtual bool should_send_page_contents() const = 0;
+  virtual mojom::APIError current_error() const = 0;
+};
```

Comment:
Why do we need this to be located in this file?

Reply:
I originally had this in ai_chat_metrics.h. i don't recall why i moved it here. i'll move it back